src/helpers/gateway/mod.rs

```diff
@@ -59,6 +60,34 @@ impl<T: Transport> Gateway<T> {
         roles: RoleAssignment,
         transport: T,
     ) -> Self {
+        let (tx, rx) = mpsc::channel();
+        let senders = Arc::new(Mutex::new(GatewaySenders::default()));
+        let senders_clone = Arc::clone(&senders);
+        let receivers = Arc::new(Mutex::new( GatewayReceivers::default()));
+        let receivers_clone = Arc::clone(&receivers);
+        thread::spawn(move|| {
+            // Perform some periodic work in the background
+            loop {
+                thread::sleep(Duration::from_secs(1));
```

Comment:
given that it is for debug purposes, timeout could be higher to avoid false positives (5-10 secs)